### PR TITLE
[pipeline](NLJ) support nested loop join for pipeline

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/pipeline/exec/nested_loop_join_probe_operator.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+class NestLoopJoinProbeOperatorBuilder final
+        : public OperatorBuilder<vectorized::VNestedLoopJoinNode> {
+public:
+    NestLoopJoinProbeOperatorBuilder(int32_t id, ExecNode* node);
+
+    OperatorPtr build_operator() override;
+};
+
+class NestLoopJoinProbeOperator final : public DataStateOperator<NestLoopJoinProbeOperatorBuilder> {
```
Comment (Gabriel39) [inline be/src/pipeline/exec/nested_loop_join_probe_operator.h:39]: I have renamed DataStateOperator to StatefulOperator

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.34 seconds
 load time: 442 seconds
 storage size: 17123356262 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221209110708_clickbench_pr_61042.html

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**